src/NUnitFramework/framework/Attributes/ValuesAttribute.cs

```diff
@@ -96,7 +96,7 @@ public ValuesAttribute(params object[] args)
         /// <summary>
         /// Get the collection of _values to be used as arguments
         /// </summary>
-        public IEnumerable GetData(ParameterInfo parameter)
+        public IEnumerable GetData(Type fixtureType, ParameterInfo parameter)
```

Comment:
Yes, those are the two main reasons. I'll post something more detailed about each use as a general comment to this PR in a short while.